custom_components/netdaemon/__init__.py

```diff
@@ -60,12 +60,16 @@ async def async_update_data():
     # Services
     async def handle_register_service(call):
         """Register custom services."""
-        daemon_class = call.data.get(ATTR_CLASS, DEFAULT_CLASS)
+        daemon_class = call.data.get(ATTR_CLASS)
         daemon_method = call.data.get(ATTR_METHOD, DEFAULT_METHOD)
+        if daemon_class:
+            service_name = f"{daemon_class}_{daemon_method}"
+        else:
+            service_name = daemon_method
 
-        LOGGER.info("Register service %s_%s", daemon_class, daemon_method)
+        LOGGER.info("Register service %s", service_name)
         hass.services.async_register(
-            DOMAIN, f"{daemon_class}_{daemon_method}", netdaemon_noop
+            DOMAIN, service_name, netdaemon_noop
```

Comment:
This is now a breaking change, why exactly is that needed?

Reply:
yea it is a breaking change. maybe we should add a new method for this instead so we do not break existing running ND versions?

Reply:
Breaking changes are fine if there is a good reason for them

Reply:
I think it is not breaking, only when class was empty we now ignore it and also do not add the _

Reply:
It's definitely breaking

Reply:
to explain, from ND the old way we use this is always send the name of the class and the method to the integration and that merges them into the name of the new service. 

In the new api the callback is not a class with a method, but just a callback delegate. This means the user should provide the service name himself and it does not make sense to have him specify two parts that get merged with an _. To make this change simple I just used the method parameter and left the class empty. The only way this is breaking is when the user called this service directly (which is not the usual scenario) AND left the class empty end expected the service to be called "no class provided_...". An alternative would be to have a third argument 'service_name' and when that is filled the integration uses that and ignores class and method